### PR TITLE
[Numpy] New FFIs for Operator: tile, trace, transpose

### PR DESCRIPTION
**Pull request by hgt312**
| Old FFI (ctypes) (us) | New FFI (cython) (us)
-- | -- | --
tile | 29.95 | 7.3
trace | 20.24 | 6.43
transpose | 22.42 | 6.07

Using 2x2 tensor as tensor input.

### CODE REVIEW

**Comment by hgt312** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by haojin2** [comment]
@hzfan Can you double check on the FFI part?

**Comment by haojin2** [comment]
@mxnet-bot  run  ci [unix-gpu]

File: src/api/operator/tensor/matrix_op.cc [hunk 1]
```diff
@@ -68,4 +68,25 @@ MXNET_REGISTER_API("_npi.clip")
...
+MXNET_REGISTER_API("_npi.tile")
+.set_body([](runtime::MXNetArgs args, runtime::MXNetRetValue* ret) {
+  using namespace runtime;
+  const nnvm::Op* op = Op::Get("_npi_tile");
+  nnvm::NodeAttrs attrs;
+  op::TileParam param;
+  if (args[1].type_code() == kDLInt) {
+    param.reps = Tuple<int>(1, args[1].operator int64_t());
+  } else {
+  param.reps = Tuple<int>(args[1].operator ObjectRef());
```
Comment (hzfan) [inline src/api/operator/tensor/matrix_op.cc:80]: Indent

**Pull request merged**